api/desecapi/models.py

```diff
@@ -183,18 +183,54 @@ def sync_from_pdns(self):
 
     @transaction.atomic
     def write_rrsets(self, rrsets):
-        q = models.Q()
+        # Always-false Q object: https://stackoverflow.com/a/35894246/6867099
+        rrsets_index = {}
```

Comment:
This variable is used to validate that no two given rrsets have the same subname and type. Is here the right place to do such validation? I believe not. Also, the database would deny writing such data. Hence, I'd prefer to move the validation somewhere else (serializer?) or drop it entirely.

Reply:
The variable was actually introduced to do a reverse lookup of the contents in the `rrsets` argument, by subname and type. It is used in https://github.com/desec-io/desec-stack/pull/80/files#diff-d93a995458c0f3a31fb831c4f43b5d27R216.

Since it was there, I figured that another validation would not hurt. Furthermore, `write_rrsets()` is a generic method of `Domain`, and not necessarily called through a serializer. In fact, it is used by the dynDNS view which is just an `APIview` without a `ModelSerializer` involved. So, there is currently no other place to put this validation. What do you think?